src/libraries/Common/src/System/Net/Http/aspnetcore/Http3/QPack/QPackDecoder.cs

```diff
@@ -294,6 +293,7 @@ private void ParseHeaderName(ReadOnlySpan<byte> data, ref int currentIndex, IHtt
             {
                 // Fast path. Store the range rather than copying.
                 _headerNameRange = (start: currentIndex, count);
+                _headerNameLength = _stringLength;
```

Comment:
This could also be inferred from the range, so I don't think it is entirely necessary here.

Reply:
As part of this bug fix, I think it's easier to follow the change if we just set the length here as you did.

In the future, we should be able to remove the length from the range field.